### PR TITLE
Barrette/products

### PR DESCRIPTION
**Pull request by BarretteBanner**

### CODE REVIEW

**Review by merecka — COMMENTED** [review]
Code looks good to me.

File: src/app/admin/product-form/product-form.component.ts [hunk 1]
```diff
@@ -1,25 +1,49 @@
...
+    private route: ActivatedRoute,
+    categoryService: CategoryService,
     private productService: ProductService
   ) {
     this.categoriesObservable = categoryService.getCategories();
+    this.id = this.route.snapshot.paramMap.get('id');
+    if (this.id)
+      this.productService
+        .get(this.id)
+        .pipe(take(1))
```
Comment (ahaly92) [inline src/app/admin/product-form/product-form.component.ts]: can you put this pipe inside of the service and use a pipe in the html? that would get rid of the need to unsubscribe which we arent doing here

File: src/app/admin/product-form/product-form.component.ts [hunk 2]
```diff
@@ -1,25 +1,49 @@
...
+        .subscribe(
+          product => {
+            this.product = product;
+          },
+          error => {
```
Comment (ahaly92) [inline src/app/admin/product-form/product-form.component.ts]: do not check in console logs

File: src/app/admin/admin-products/admin-products.component.ts [hunk 3]
```diff
@@ -1,7 +1,33 @@
...
+import { Product } from 'src/app/models/product';
 
 @Component({
   selector: 'app-admin-products',
   templateUrl: './admin-products.component.html'
 })
-export class AdminProductsComponent {}
+export class AdminProductsComponent implements OnDestroy {
+  productsObservable = [];
+  filteredProducts: any[];
```
Comment (ahaly92) [inline src/app/admin/admin-products/admin-products.component.ts]: we should create a model for product. using any defeats the point of using typescript. Generally you wanna avoid using any unless you have to

File: src/app/admin/admin-products/admin-products.component.ts [hunk 4]
```diff
@@ -1,7 +1,33 @@
...
   selector: 'app-admin-products',
   templateUrl: './admin-products.component.html'
 })
-export class AdminProductsComponent {}
+export class AdminProductsComponent implements OnDestroy {
+  productsObservable: Product[] = [];
+  filteredProducts: Product[];
+  subscription: Subscription;
+  constructor(productService: ProductService) {
+    this.subscription = productService
```
Comment (ahaly92) [inline src/app/admin/admin-products/admin-products.component.ts]: subscription -> productsSubscription for readability

**Review by ahaly92 — APPROVED** [review]
one small change! This is really great stuff!!

**Pull request merged**